src/proxy/resolve.rs

```diff
@@ -58,13 +60,29 @@ pub struct MakeSvc<R, M> {
 
 /// Observes an `R`-typed resolution stream, using an `M`-typed endpoint stack to
 /// build a service for each endpoint.
-#[derive(Clone, Debug)]
-pub struct Discover<R, M> {
+pub struct Discover<R: Resolution, M: svc::Service<R::Endpoint>> {
     resolution: R,
     make: M,
+    makes: MakeStream<M::Future>,
     is_empty: Arc<AtomicBool>,
 }
 
+struct MakeStream<F> {
+    futures: FuturesUnordered<MakeFuture<F>>,
+    cancelations: IndexMap<SocketAddr, oneshot::Sender<()>>,
```

Comment:
Typo in field name

Reply:
I _definitely_ spelled `cancellations` right --- it looks like a80396b6c17f1e4e871bb252488e31582c7b61b0 broke this :P

Reply:
Both spellings are correct

Reply:
Notice, for instance that we use "Canceled" elsewhere in this file. A quick search says that the single el is an Americanism. Let it henceforth be policy that we use Americanized spelling in Linkerd.

Reply:
TIL that single-el is valid!